org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/wizard/template/NewProjectWizardTemplateParameterPage.java

```diff
@@ -0,0 +1,165 @@
+/*******************************************************************************
+ * Copyright (c) 2017 itemis AG (http://www.itemis.de) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.xtext.ui.wizard.template;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.jface.dialogs.IMessageProvider;
+import org.eclipse.jface.wizard.WizardPage;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.SelectionAdapter;
+import org.eclipse.swt.events.SelectionEvent;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.Combo;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Control;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.swt.widgets.Text;
+import org.eclipse.swt.widgets.Widget;
+import org.eclipse.xtext.util.Pair;
+import org.eclipse.xtext.util.Tuples;
+
+/**
+ * Page in the new project wizard to display a list with project templates. User get some description text and can select a template. In a
+ * later page he can eventually configure the project template further.
+ * 
+ * @author Arne Deutsch - Initial contribution and API
+ * @since 2.14
+ */
+public class NewProjectWizardTemplateParameterPage extends WizardPage {
+
+	private final AbstractProjectTemplate projectTemplate;
+	private final List<Pair<ProjectVariable, Control>> variableWidgets = new ArrayList<>();
```

Comment:
Personally I don't like `List<Pair<..>>`

Reply:
Alternatice would be to introduce an extra class that bundles a ProjectVariable with a Control. As it is a minor implementation detail that can be changed easaly at any time I would propose to stick to the Pair for now.

Reply:
Ok

Reply:
Actually this will vanish with my next comit. Refernce is then hold directly in the ProjectVariable, no Pair needed any more.